### ISSUE TITLE
The --internal flag in podman network create disables dns names

### ISSUE THREAD

**Issue by catric96** [containers/dnsname#51]
Environment:
Podman 2.2.1 running on RHEL 8.2
```
Version:      2.2.1
API Version:  2.1.0
Go Version:   go1.14.7
Built:        Thu Dec 10 13:26:48 2020
```

Steps to reproduce:
Create two networks via `podman network create,` one of which has the `--internal` flag set. Then, test each network created by attaching it to two containers and attempting to ping back and forth by their respective names. Note that you cannot get the containers to ping each other by name when using the network with `--internal` set.

When running `podman network inspect` on each of the two network interfaces I can see that the interface with `--internal` has `"isGateway"` set to false, while the one without the flag has it set to true. Both still contain the `dnsname` plugin in their definition. 

TL;DR:
Would it be possible to have dnsname be functional even without a gateway? Or does there exist another way to restrict external access for containers without `--internal` set?

**Comment by baude** [comment]
the gateway is probably required ... it is being used today as the dns server for each container.  I'll look into the podman side to see if I can restrict dnsname from being added when --internal is set.
